pinot-broker/src/test/java/com/linkedin/pinot/broker/queryquota/TableQueryQuotaManagerTest.java

```diff
@@ -0,0 +1,351 @@
+    return new FakeHelixManager(helixClusterName, helixClusterName + "_1", InstanceType.CONTROLLER, ZkStarter.DEFAULT_ZK_STR);
+  }
+
+  public class FakeHelixManager extends ZKHelixManager {
+    private ZkHelixPropertyStore<ZNRecord> _propertyStore;
+
+    FakeHelixManager(String clusterName, String instanceName, InstanceType instanceType, String zkAddress) {
+      super(clusterName, instanceName, instanceType, zkAddress);
+      super._zkclient = new ZkClient(StringUtil.join("/", StringUtils.chomp(ZkStarter.DEFAULT_ZK_STR, "/")),
+          ZkClient.DEFAULT_SESSION_TIMEOUT, ZkClient.DEFAULT_CONNECTION_TIMEOUT, new ZNRecordSerializer());
+      _zkclient.deleteRecursive("/" + clusterName + "/PROPERTYSTORE");
+      _zkclient.createPersistent("/" + clusterName + "/PROPERTYSTORE", true);
+      setPropertyStore(clusterName);
+    }
+
+    void setPropertyStore(String clusterName) {
+      _propertyStore = new ZkHelixPropertyStore<>(new ZkBaseDataAccessor<ZNRecord>(_zkclient),
+          "/" + clusterName + "/PROPERTYSTORE", null);
+    }
+
+    void closeZkClient() {
+      _zkclient.close();
+    }
+  }
+
+  @AfterMethod
+  public void afterMethod() {
+    if (_helixManager instanceof FakeHelixManager) {
+      _testPropertyStore.reset();
+      ZKMetadataProvider.removeResourceConfigFromPropertyStore(_testPropertyStore, OFFLINE_TABLE_NAME);
+      ZKMetadataProvider.removeResourceConfigFromPropertyStore(_testPropertyStore, REALTIME_TABLE_NAME);
+    }
+    _tableQueryQuotaManager.cleanUpRateLimiterMap();
+  }
+
+  @AfterTest
+  public void afterTest() {
+    if (_helixManager instanceof FakeHelixManager) {
+      ((FakeHelixManager) _helixManager).closeZkClient();
+    }
+    ZkStarter.stopLocalZkServer(_zookeeperInstance);
+  }
+
+  @Test
+  public void testOfflineTableNotnullQuota() throws Exception {
+    ExternalView brokerResource = generateBrokerResource(OFFLINE_TABLE_NAME);
+    TableConfig tableConfig = generateDefaultTableConfig(OFFLINE_TABLE_NAME);
+    setQps(tableConfig);
+    _tableQueryQuotaManager.initTableQueryQuota(tableConfig, brokerResource);
+    Assert.assertEquals(_tableQueryQuotaManager.getRateLimiterMapSize(), 1);
+
+    _tableQueryQuotaManager.dropTableQueryQuota(tableConfig, brokerResource);
+    Assert.assertEquals(_tableQueryQuotaManager.getRateLimiterMapSize(), 0);
+  }
+
+  @Test
+  public void testOfflineTableWithNullQuotaAndNoRealtimeTableConfig() throws Exception {
+    ExternalView brokerResource = generateBrokerResource(OFFLINE_TABLE_NAME);
+    TableConfig tableConfig = generateDefaultTableConfig(OFFLINE_TABLE_NAME);
+    _tableQueryQuotaManager.initTableQueryQuota(tableConfig, brokerResource);
+    Assert.assertEquals(_tableQueryQuotaManager.getRateLimiterMapSize(), 0);
+  }
+
+  @Test
+  public void testOfflineTableWithNullQuotaButWithRealtimeTableConfigNullQpsConfig() throws Exception {
+    ZNRecord znRecord = new ZNRecord("broker_instance_1");
```

Comment:
You probably want to build the table configuration here, check `TableConfig.Builder`